dart/dynamics/ShapeFrame.cpp

```diff
@@ -0,0 +1,394 @@
+VisualDataProperties::VisualDataProperties(const Eigen::Vector4d& color,
+                                           const bool hidden)
+  : mRGBA(color),
+    mHidden(hidden)
+{
+  // Do nothing
+}
+
+//==============================================================================
+CollisionDataProperties::CollisionDataProperties(
+    const bool collidable)
+  : mCollidable(collidable)
+{
+  // Do nothing
+}
+
+//==============================================================================
+DynamicsDataProperties::DynamicsDataProperties(
+    const double frictionCoeff,
+    const double restitutionCoeff)
+  :
+    mFrictionCoeff(frictionCoeff),
+    mRestitutionCoeff(restitutionCoeff)
+{
+  // Do nothing
+}
+
+} // namespace detail
+
+//==============================================================================
+VisualAddon::VisualAddon(common::AddonManager* mgr,
+                         const PropertiesData& properties)
+  : AddonWithProtectedProperties<
+      VisualAddon,
+      detail::VisualDataProperties,
+      ShapeFrame,
+      &detail::VisualShapeFrameAddonUpdate>(mgr, properties),
+    mShapeFrame(dynamic_cast<ShapeFrame*>(mgr))
+{
+  assert(mShapeFrame);
+}
+
+//==============================================================================
+void VisualAddon::setRGBA(const Eigen::Vector4d& color)
+{
+  mProperties.mRGBA = color;
+
+  UpdateProperties(this);
+
+  mShapeFrame->getShape()->notifyColorUpdate(color);
+}
+
+//==============================================================================
+void VisualAddon::setColor(const Eigen::Vector3d& color)
+{
+  setRGB(color);
+}
+
+//==============================================================================
+void VisualAddon::setColor(const Eigen::Vector4d& color)
+{
+  setRGBA(color);
+}
+
+//==============================================================================
+void VisualAddon::setRGB(const Eigen::Vector3d& rgb)
+{
+  Eigen::Vector4d rgba = getRGBA();
+  rgba.head<3>() = rgb;
+
+  setRGBA(rgba);
+}
+
+//==============================================================================
+void VisualAddon::setAlpha(const double alpha)
+{
+  Eigen::Vector4d rgba = getRGBA();
+  rgba[3] = alpha;
+
+  setRGBA(rgba);
+}
+
+//==============================================================================
+Eigen::Vector3d VisualAddon::getColor() const
+{
+  return getRGB();
+}
+
+//==============================================================================
+Eigen::Vector3d VisualAddon::getRGB() const
+{
+  return getRGBA().head<3>();
+}
+
+//==============================================================================
+const double VisualAddon::getAlpha() const
+{
+  return getRGBA()[3];
+}
+
+//==============================================================================
+void VisualAddon::hide()
+{
+  setHidden(false);
+}
+
+//==============================================================================
+void VisualAddon::show()
+{
+  setHidden(true);
```

Comment:
`show()` and `hide()` appear to be reversed - unless `setHidden` has very unintuitive behavior. :smile: